lib/logstash/codecs/cef.rb

```diff
@@ -72,102 +76,10 @@ class LogStash::Codecs::CEF < LogStash::Codecs::Base
-      "cn2" => "deviceCustomNumber2",
-      "cn2Label" => "deviceCustomNumber2Label",
-      "cn3" => "deviceCustomNumber3",
-      "cn3Label" => "deviceCustomNumber3Label",
-      "cnt" => "baseEventCount",
-      "cs1" => "deviceCustomString1",
-      "cs1Label" => "deviceCustomString1Label",
-      "cs2" => "deviceCustomString2",
-      "cs2Label" => "deviceCustomString2Label",
-      "cs3" => "deviceCustomString3",
-      "cs3Label" => "deviceCustomString3Label",
-      "cs4" => "deviceCustomString4",
-      "cs4Label" => "deviceCustomString4Label",
-      "cs5" => "deviceCustomString5",
-      "cs5Label" => "deviceCustomString5Label",
-      "cs6" => "deviceCustomString6",
-      "cs6Label" => "deviceCustomString6Label",
-      "dhost" => "destinationHostName",
-      "dmac" => "destinationMacAddress",
-      "dntdom" => "destinationNtDomain",
-      "dpid" => "destinationProcessId",
-      "dpriv" => "destinationUserPrivileges",
-      "dproc" => "destinationProcessName",
-      "dpt" => "destinationPort",
-      "dst" => "destinationAddress",
-      "duid" => "destinationUserId",
-      "duser" => "destinationUserName",
-      "dvc" => "deviceAddress",
-      "dvchost" => "deviceHostName",
-      "dvcpid" => "deviceProcessId",
-      "end" => "endTime",
-      "fname" => "fileName",
-      "fsize" => "fileSize",
-      "in" => "bytesIn",
-      "msg" => "message",
-      "out" => "bytesOut",
-      "outcome" => "eventOutcome",
-      "proto" => "transportProtocol",
-      "request" => "requestUrl",
-      "rt" => "deviceReceiptTime",
-      "shost" => "sourceHostName",
-      "smac" => "sourceMacAddress",
-      "sntdom" => "sourceNtDomain",
-      "spid" => "sourceProcessId",
-      "spriv" => "sourceUserPrivileges",
-      "sproc" => "sourceProcessName",
-      "spt" => "sourcePort",
-      "src" => "sourceAddress",
-      "start" => "startTime",
-      "suid" => "sourceUserId",
-      "suser" => "sourceUserName",
-      "ahost" => "agentHostName",
-      "art" => "agentReceiptTime",
-      "at" => "agentType",
-      "aid" => "agentId",
-      "_cefVer" => "cefVersion",
-      "agt" => "agentAddress",
-      "av" => "agentVersion",
-      "atz" => "agentTimeZone",
-      "dtz" => "destinationTimeZone",
-      "slong" => "sourceLongitude",
-      "slat" => "sourceLatitude",
-      "dlong" => "destinationLongitude",
-      "dlat" => "destinationLatitude",
-      "catdt" => "categoryDeviceType",
-      "mrt" => "managerReceiptTime",
-      "amac" => "agentMacAddress"
-  }
-
-  # Reverse mapping of CEF full field names to CEF extensions field names for encoding into a CEF event for output.
-  REVERSE_MAPPINGS = MAPPINGS.invert
+  # Defines whether a set of device-specific CEF fields represent the _observer_,
+  # or the actual `host` on which the event occurred. If this codec handles a mix,
+  # it is safe to use the default `observer`.
+  config :device, :validate => %w(observer host), :default => 'observer'
```

Comment:
The more I read into the ECS conventions, the more I think `host` should be the default. From the ECS docs (emphasis mine)

 - `observer`: a "_**special** network, security, or application device used to detect, observe, or create network, security, or application-related events and metrics_"
 - `host`: "_host on which the event happened, **or** from which the measurement was taken_"

Since `observer` is defined as _special_, to me that indicates that users should be able to opt-in when they align with the special case.

If the default changes, we also need to adjust the docs to better describe the situations in which a user would override the default.

Reply:
I think it's a bit more also about what kind of data is usually ingested, from my own experience the amount of logsources that usually is used together with ArcSight (outside of Windows which would be host.*) is network devices, proxies, load balancers, IDS/IPS etc, mostly related to an "observer" role and field in ECS.

I think that as long as we make it easy for the end-user to change or modify it then we should be fine, but the "best" default will always depend on the customers usecase unfortunately.

Reply:
The renaming of these fields could be done by `logstash-filter-mutate` instead of provide a setting to this plugin to say "host" or "observer" since this decision could be made at runtime by some login in the user's pipeline. For example he could use a tag to mark Windows originated events so, in that case `host` is correct but for other events it should be `observer`.
If in the majority of the cases the `observer` is the best choice I would go with that, perhaps suggesting in the docs that it could be renamed with a subsequent filter.

Reply:
@P1llus do you agree on the thoughts above? So that we can go with `observer` and let eventually the user to rename to `host` ?

Reply:
That sounds like a logical conclusion yes @andsel . We might want to define that somewhere in the documentation though maybe?

Reply:
yes where we document the `device` setting we say that by default is `observer` but user can also select `host`

Reply:
it's already documented in https://github.com/logstash-plugins/logstash-codec-cef/pull/83/files#diff-cae5619b3d18ec99c5ccd0a9f6de0c6d3f53343c64692444551a7d29da6863e7R96-R101